pytensor/link/numba/dispatch/scalar.py

```diff
@@ -127,20 +132,18 @@ def {scalar_op_fn_name}({", ".join(input_names)}):
     return direct_cast(scalar_func_numba({converted_call_args}, np.intc(1)), output_dtype)
             """
 
-    scalar_op_fn = compile_function_src(
-        scalar_op_src, scalar_op_fn_name, {**globals(), **global_env}
+    scalar_op_fn = compile_numba_function_src(
+        scalar_op_src,
+        scalar_op_fn_name,
+        {**globals(), **global_env},
     )
 
-    signature = create_numba_signature(node, force_scalar=True)
+    # Functions that call a function pointer can't be cached
+    cache_key = None if cython_func else scalar_op_cache_key(op)
```

Comment:
I thought this wasn't true? Can't we cache the linalg stuff? I recall a conversation with one of the numba devs about this.